src/odb/include/odb/db.h

```diff
@@ -3503,6 +3504,10 @@ class dbVia : public dbObject
   //
   dbVia* getBlockVia();
 
+  void setAddedByRouter(bool);
+
+  bool isAddedByRouter();
```

Comment:
In pin access, we need to generate vias in tritonroute if no default vias available. We then write these vias into opendb and reference them in dbAccessPoints. If we want to do routing after that without pin access, we read these default vias from opendb along with the access points that reference them.

Reply:
I only see it being used for
      if (via->isAddedByRouter())
        viaDef->setDefault(true);

so why not just set the via as a default directly instead of using this flag to set it later?

Reply:
If you mean set the dbVia as default via, dbVia does not have a default flag (dbTechVia has). If you mean to set all the block vias as default vias, what about vias added by other tools that should not be used as default?

Reply:
What other tool creates default vias?  pdngen generates vias for the power grid but they should not be default.

"Default vias are used for default routing by the **signal routers**."